schema/query/projection_evaluator.go

```diff
@@ -135,12 +136,12 @@ func evalProjectionArray(ctx context.Context, pf ProjectionField, payload []inte
 			}
 			res = append(res, v)
 		}
-	} else if array, ok := validator.(*schema.Array); ok {
+	case *schema.Array:
```

Comment:
Since `*schema.Array` implements `FieldGetter`, the `case *schema.Array` need to be placed first or we would always end up at the FieldGetter case and never hit the `*schema.Array` case.

Reply:
https://play.golang.org/p/nYdtczbVxYJ